concent_api/core/views.py

```diff
@@ -1220,6 +1225,632 @@ def store_message_and_message_status(
+    reject_report_computed_task:    message.RejectReportComputedTask     = None,
+    subtask_results_accepted:       message.tasks.SubtaskResultsAccepted = None,
+    subtask_results_rejected:       message.tasks.SubtaskResultsRejected = None,
+):
+    try:
+        subtask = Subtask.objects.get(
+            subtask_id = subtask_id,
+        )
+    except Subtask.DoesNotExist:
+        subtask = None
+
+    if subtask is not None:
+        subtask = update_subtask(
+            subtask                         = subtask,
+            state                           = state,
+            next_deadline                   = next_deadline,
+            set_next_deadline               = set_next_deadline,
+            task_to_compute                 = task_to_compute,
+            report_computed_task            = report_computed_task,
+            ack_report_computed_task        = ack_report_computed_task,
+            reject_report_computed_task     = reject_report_computed_task,
+            subtask_results_accepted        = subtask_results_accepted,
+            subtask_results_rejected        = subtask_results_rejected,
+        )
+    else:
+        subtask = store_subtask(
+            task_id                         = task_id,
+            subtask_id                      = subtask_id,
+            provider_public_key             = provider_public_key,
+            requestor_public_key            = requestor_public_key,
+            state                           = state,
+            next_deadline                   = next_deadline,
+            task_to_compute                 = task_to_compute,
+            report_computed_task            = report_computed_task,
+            ack_report_computed_task        = ack_report_computed_task,
+            reject_report_computed_task     = reject_report_computed_task,
+            subtask_results_accepted        = subtask_results_accepted,
+            subtask_results_rejected        = subtask_results_rejected,
+        )
+    return subtask
+
+
+def store_message(
+    golem_message:          message.base.Message,
+    task_id:                str,
+    subtask_id:             str,
+):
+    assert golem_message.TYPE in message.registered_message_types
+
+    message_timestamp = datetime.datetime.now(timezone.utc)
+    stored_message = StoredMessage(
+        type        = golem_message.TYPE,
+        timestamp   = message_timestamp,
+        data        = copy.copy(golem_message).serialize(),
+        task_id     = task_id,
+        subtask_id  = subtask_id,
+    )
+    stored_message.full_clean()
+    stored_message.save()
+
+    return stored_message
+
+
+def create_file_transfer_token(
+    report_computed_task: message.tasks.ReportComputedTask,
+    client_public_key: bytes,
+) -> message.concents.FileTransferToken:
+    """
+    Function to create FileTransferToken from ReportComputedTask message
+    to check existence of a file on cluster storage
```

Comment:
This is not true. I see this function used when creating tokens for any purpose. Not only to check file existence but also to give it to the client so that it can download or upload its files.

Reply:
Done